// @generated/type-graphql/resolvers/relations/Exercise/ExerciseRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExerciseRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Exercise_1 = require("../../../models/Exercise");
const Group_1 = require("../../../models/Group");
const Workout_1 = require("../../../models/Workout");
const ExerciseGroupArgs_1 = require("./args/ExerciseGroupArgs");
const ExerciseWorkoutArgs_1 = require("./args/ExerciseWorkoutArgs");
const helpers_1 = require("../../../helpers");
let ExerciseRelationsResolver = class ExerciseRelationsResolver {
    async group(exercise, ctx, info, args) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).exercise.findUniqueOrThrow({
            where: {
                id: exercise.id,
            },
        }).group({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async Workout(exercise, ctx, info, args) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).exercise.findUniqueOrThrow({
            where: {
                id: exercise.id,
            },
        }).Workout({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
};
exports.ExerciseRelationsResolver = ExerciseRelationsResolver;
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Group_1.Group, {
        nullable: true
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__param(3, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Exercise_1.Exercise, Object, Object, ExerciseGroupArgs_1.ExerciseGroupArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], ExerciseRelationsResolver.prototype, "group", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Workout_1.Workout], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__param(3, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Exercise_1.Exercise, Object, Object, ExerciseWorkoutArgs_1.ExerciseWorkoutArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], ExerciseRelationsResolver.prototype, "Workout", null);
exports.ExerciseRelationsResolver = ExerciseRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Exercise_1.Exercise)
], ExerciseRelationsResolver);
